web/modules/custom/asset_release/asset_release.module

```diff
@@ -91,3 +94,84 @@ function asset_release_theme($existing, $type, $theme, $path) {
     ],
   ];
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ *
+ * Interrupts the permissions for updating a release. The permission is not
+ * granted if there are no transitions allowed for the current state of the
+ * node.
+ */
+function asset_release_rdf_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  if ($entity->bundle() != 'asset_release' || $account->isAnonymous() || \Drupal::currentUser()->hasPermission('administer rdf entity')) {
+    return AccessResult::neutral();
+  }
+
+  $parent = \Drupal::service('asset_release.relations')->getReleaseSolution($entity);
+
+  switch ($operation) {
+    case 'view':
+      $membership = Og::getMembership($parent, $account);
+      if (empty($membership)) {
+        return AccessResult::neutral();
+      }
+      if (!$entity->isPublished() && $membership->hasPermission('view unpublished rdf entity')) {
+        return AccessResult::allowed();
+      }
+      if ($entity->isPublished() && $membership->hasPermission('view rdf entity')) {
+        return AccessResult::allowed();
+      }
+      break;
+
+    case 'edit':
+      $allowed_transitions = $entity->field_isr_state->first()->getTransitions();
+      return AccessResult::forbiddenIf(empty($allowed_transitions));
+
+    default:
+      break;
+  }
+  return AccessResult::neutral();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * Default graph is published. Draft graph is draft.
+ */
+function asset_release_rdf_entity_presave(EntityInterface $entity) {
```

Comment:
If I fix this now, I will have to take care of the backwards compatibility (backport it to collections solutions and news). Is this acceptable.?